### ISSUE TITLE
RandomLinkSplit can create overlapping negative val and test edges for undirected graphs

### ISSUE THREAD

**Issue by notableTieView** [pyg-team/pytorch_geometric#10706]
### 🐛 Describe the bug

Using RandomLinkSplit on an **undirected** graph can cause an overlap between negative edge samples in val and test, i.e. a negative edge (a,b) can be split into a negative sample (a,b) in val and a negative sample (b,a) in test. Since the graph is undirected, that is the same negative sample in both sets, thus overlap between val and test set.

```
import torch
import torch_geometric
from torch_geometric import seed_everything
from torch_geometric.data import Data
from torch_geometric.utils import to_undirected
from torch_geometric.transforms import RandomLinkSplit

# V={0,1,2,3}
# E={(0,1),(1,2),(2,3)}
undirected_edges = to_undirected(
    torch.tensor([
        [0,1,2],
        [1,2,3]
    ])
)
data = Data(edge_index=undirected_edges, num_nodes=4)
print("Graph is undirected: ", data.is_undirected())

seed_everything(1)
transform = RandomLinkSplit(
    is_undirected=True, 
    num_val=0.4, num_test=0.4,
    add_negative_train_samples=False
)
train_data, val_data, test_data = transform(data)

print("val edges:\n", val_data.edge_label_index)
print("test edges:\n", test_data.edge_label_index)

# Output:
# Graph is undirected:  True
# val edges:
# tensor([[2, 0],
#         [3, 3]])
# test edges:
# tensor([[0, 3],
#         [1, 0]])
```

In the example, the negative sample is (0,3) is shared as (0,3) in val and (3,0) in test.
The issue seems to come from ignoring the `is_undirected` param in the call to `negative_sampling` in `RandomLinkSplit.forward()`.

### Versions

```
aiohappyeyeballs        2.6.2
aiohttp                 3.13.5
aiosignal               1.4.0
appnope                 0.1.4
asttokens               3.0.1
attrs                   26.1.0
certifi                 2026.5.20
charset-normalizer      3.4.7
comm                    0.2.3
contourpy               1.3.3
cycler                  0.12.1
debugpy                 1.8.20
decorator               5.3.1
executing               2.2.1
filelock                3.29.0
fonttools               4.63.0
frozenlist              1.8.0
fsspec                  2026.4.0
idna                    3.17
ipykernel               7.2.0
ipython                 9.14.0
ipython_pygments_lexers 1.1.1
ipywidgets              8.1.8
jedi                    0.20.0
Jinja2                  3.1.6
joblib                  1.5.3
jupyter_client          8.8.0
jupyter_core            5.9.1
jupyterlab_widgets      3.0.16
kiwisolver              1.5.0
MarkupSafe              3.0.3
matplotlib              3.10.9
matplotlib-inline       0.2.2
mpmath                  1.3.0
multidict               6.7.1
nest-asyncio            1.6.0
networkx                3.6.1
numpy                   2.4.6
packaging               26.2
parso                   0.8.7
pexpect                 4.9.0
pillow                  12.2.0
pip                     25.3
platformdirs            4.10.0
prompt_toolkit          3.0.52
propcache               0.5.2
psutil                  7.2.2
ptyprocess              0.7.0
pure_eval               0.2.3
pyg_lib                 0.6.0+pt211
Pygments                2.20.0
pyparsing               3.3.2
python-dateutil         2.9.0.post0
pyzmq                   27.1.0
requests                2.34.2
scikit-learn            1.8.0
scipy                   1.17.1
setuptools              81.0.0
six                     1.17.0
stack-data              0.6.3
sympy                   1.14.0
threadpoolctl           3.6.0
torch                   2.11.0
torch-geometric         2.7.0
torch_scatter           2.1.2
torch_sparse            0.6.18
tornado                 6.5.6
tqdm                    4.67.3
traitlets               5.15.0
typing_extensions       4.15.0
urllib3                 2.7.0
wcwidth                 0.7.0
widgetsnbextension      4.0.15
xxhash                  3.7.0
yarl                    1.24.2
```

**Comment by TasfinMahmud** [comment]
Hi! I'd like to take a look at this if no one else is working on it. The root cause seems to be that RandomLinkSplit doesn't canonicalize edge direction when checking for overlap in undirected graphs — so (a, b) in val and (b, a) in test are treated as distinct edges even though they represent the same undirected connection. I'll investigate and submit a fix.

**Comment by TasfinMahmud** [comment]
hey! I ran into something similar while working on link prediction for a collab filtering project — the undirected edge symmetry thing is super sneaky. I'd love to take a crack at this if nobody else is on it 🙌
